src/neighborsearch/kball_searcher.jl

```diff
@@ -0,0 +1,52 @@
+# ------------------------------------------------------------------
+# Licensed under the ISC License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+KBallSearcher(object, neighborhood)
+
+A method for searching neighbors in spatial `object` inside `neighborhood`.
+Used when doing estimation combining k-nearest and radius.
+It searches first for the k-nearest by using the internal kdtree and then
+limiting the results within the `BallNeighborhood` by calling `isneighbor`
+"""
+struct KBallSearcher{S<:AbstractNeighborSearcher} <: AbstractBoundedNeighborSearcher
+  searcher::S
+  nmax::Int
+end
+
+object(searcher::KBallSearcher) = object(searcher.searcher)
```

Comment:
I think this isn't needed if you just call the `search` method of the inner `searcher` directly. See comment bellow.